provisioning/device/test/_polling_state_machine_test.js

```diff
@@ -263,27 +269,22 @@ describe('state machine', function () {
     /* Tests_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_025: [ If `cancel` is called while disconnected, it shall immediately call its `callback`. ] */
     it ('does nothing if called while disconnected', function (testCallback) {
       machine.cancel(function(err) {
-        assert(!err);
+        assert.isNotAnError(err);
         assert(machine);
         assertNoTransportFunctionsCalled(machine);
         testCallback();
       });
     });
 
-    describe('calls cancel', function () {
-
-      /* Tests_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_026: [ `cancel` shall call `cancel` of it's called while the transport is connected. ] */
-      it ('if called while connected', function (testCallback) {
-        callRegisterWithDefaultArgs(function(err) {
-          assert(!err);
-          assert.isFalse(machine._transport.cancel.called);
-          machine.cancel(function(err) {
-            assert(!err);
-            assert(machine._transport.cancel.calledOnce);
-            testCallback();
-          });
-        });
+    /* Tests_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_030: [ If `cancel` is called while the transport is connected but idle, it shall immediately call its `callback`. ] */
+    it ('does nothing if called while idle', function (testCallback) {
+      callRegisterWithDefaultArgs(function(err) {
+        assert.isNotAnError(err);
+        testCallback();
```

Comment:
>testCallback [](start = 8, length = 12)

you're not calling cancel anymore?

Reply:
Not if the transport is connected but idle.  It used to be that 'cancel' meant disconnect in which case it made sense.  now that 'cancel' means 'cancel the current operation', it doesn't make sense to call it anymore.

---
In reply to: [160014792](https://github.com/Azure/azure-iot-sdk-node/pull/192#discussion_r160014792) [](ancestors = 160014792)